lib/javascript/JavascriptParser.js

```diff
@@ -2627,30 +2635,88 @@ class JavascriptParser extends Parser {
 
 	/**
 	 * @param {ObjectPattern} objectPattern object pattern
-	 * @returns {Set<string> | undefined} set of names or undefined if not all keys are identifiers
+	 * @returns {Set<DestructuringAssignmentProperty> | undefined} set of names or undefined if not all keys are identifiers
 	 */
 	_preWalkObjectPattern(objectPattern) {
-		const ids = new Set();
+		/** @type {Set<DestructuringAssignmentProperty>} */
+		const props = new Set();
 		const properties = objectPattern.properties;
 		for (let i = 0; i < properties.length; i++) {
 			const property = properties[i];
 			if (property.type !== "Property") return;
+			if (property.shorthand && property.value.type === "Identifier") {
+				this.scope.inShorthand = property.value.name;
+			}
 			const key = property.key;
-			if (key.type === "Identifier") {
-				ids.add(key.name);
+			if (!property.computed && key.type === "Identifier") {
+				const pattern =
```

Comment:
I would suggest to simplify and bail out if there is no identifiers on right side of computed property. Problem is that on this stage we dont know is this import specifier in assignment expression and parsing all code is useless in 99% of cases.

I would suggest to stick with simple cases like `{ a, b: c } = d`

wdyt @alexander-akait